src/components/AddMovieDialog/AddMovieDialog.jsx

```diff
@@ -1,15 +1,27 @@
 import React, {useCallback} from 'react';
 import ModalDialog from '../Common/ModalDialog';
 import MovieForm from '../Common/MovieForm'
+import { connect } from 'react-redux'
+import {commonActions, movieActions} from '../../store/actions'
 
-const AddMovieDialog = ({onClose}) => {
+const AddMovieDialog = ({onClose, onConfirm, genres}) => {
     const onDialogClose = useCallback(() => onClose(), [onClose]);
+    const onSubmit = useCallback((data) => onConfirm(data), [onConfirm]);
```

Comment:
not sure you need this

just use `onConfirm` ins props `onSubmit={onConfirm}`

Reply:
On one of previous meetings we discussed it.  And I decided that if we pass the function in child component we have to use useCallback. I can be wrong.